
#######################################################################################################
                             PROJECT: Thefuck
#######################################################################################################
CLONING PROJECT:
✓ SUCCESSFUL! This project is already cloned: https://github.com/nvbn/thefuck
.......................................................................................................
BUGGY LINES:
Suspicious lines to bug1:
15,17
Suspicious lines to bug2:
121
Suspicious lines to bug3:
108,110
Suspicious lines to bug4:
23,24,25
Suspicious lines to bug5:
9
Suspicious lines to bug6:
10,17
Suspicious lines to bug7:
6
Suspicious lines to bug8:
18,19,20,21,29
Suspicious lines to bug9:
27
Suspicious lines to bug10:
23,32
Suspicious lines to bug11:
15
Suspicious lines to bug12:
3,9
Suspicious lines to bug13:
9,10
Suspicious lines to bug14:
10,11,12,13,14
Suspicious lines to bug15:
8,9,15,17
Suspicious lines to bug16:
9,10,11,10,12
Suspicious lines to bug17:
12,27,29,30,31
Suspicious lines to bug18:

Suspicious lines to bug19:
15
Suspicious lines to bug20:
16,31
Suspicious lines to bug21:
8,9
Suspicious lines to bug22:
84,85,86
Suspicious lines to bug23:
183
Suspicious lines to bug24:
57,58,64,68,70
Suspicious lines to bug25:
13
Suspicious lines to bug26:
10,13
Suspicious lines to bug27:
26
Suspicious lines to bug28:
23,26,28,64,66
Suspicious lines to bug29:
26,27,28
Suspicious lines to bug30:
53
Suspicious lines to bug31:
13
Suspicious lines to bug32:
2
.......................................................................................................

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 1

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 2ced7a7f33ae0bec3ffc7a43ce95330bdf6cfcb9
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/pip_unknown_command.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_pip_unknown_command.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/pip_unknown_command.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [15, 17]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 15
x Fail! Line: 15 could not be transformed: 'broken_cmd = re.findall(r'ERROR unknown command \"([a-z]+)\"','
Start transforming line: 17
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 2 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[15]
✓ Number of created sketches:2
Time needed: 0.02 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 2

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 40ab4eb62db57627bff10cf029d29c94704086a2
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/utils.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/test_utils.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/utils.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [121]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 121
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 3

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: ac343fb1bd7dadbb9e1a9fd7e3071f5778e338a4
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/shells/test_fish.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [108, 110]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 108
x Fail! Line: 108 could not be transformed: 'proc = Popen(['fish', '-c', 'echo $FISH_VERSION'],'
Start transforming line: 110
x Fail! Line: 110 could not be transformed: 'version = proc.stdout.read().decode('utf-8').strip()'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[108, 110]
✓ Number of created sketches:0
Time needed: 0.02 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 4

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 68949a592248913b52bfd50036893553153fcddb
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/shells/test_fish.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [23, 24, 25]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 23
x Fail! Line: 23 could not be transformed: 'alias_out = proc.stdout.read().decode('utf-8').strip().split('\n')'
Start transforming line: 24
✓
Start transforming line: 25
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 54 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[23]
✓ Number of created sketches:54
Time needed: 0.02 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 5

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 7c858fadb3458be829d3d43666ccb46c3ed5b8a0
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_push.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [9]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 9
x Fail! Line: 9 could not be transformed: 'and 'set-upstream' in command.output)'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[9]
✓ Number of created sketches:0
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 6

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 797ca1c5647c565f62e21a8e29515c8b0fbe275f
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_branch_exists.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_branch_exists.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_branch_exists.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [10, 17]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 10
x Fail! Line: 10 could not be transformed: 'and " already exists." in command.output)'
Start transforming line: 17
x Fail! Line: 17 could not be transformed: 'r"fatal A branch named '([^']*)' already exists.", command.output)[0]'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[10, 17]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 7

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 64d6835e15a22cf8803d398cbb593f748c550e8a
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/php_s.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_php_s.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/php_s.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [6]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 6
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 3 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:3
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 8

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 449cb9a00693c8b4d97d5fda8d732cf0978e117e
x FAIL! The TEST file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_dnf_no_such_command.py to the project Thefuck with bug: 8does not exists. Next bug is taken

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 9

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: ce6b82c92d78ae283cb3db001766b76f6647bc47
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_push.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [27]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 27
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 2 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:2
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 10

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 8bd6c5da67e55c64257345efa4e3cc454c42475c
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/man.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_man.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/man.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [23, 32]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 23
x Fail! Line: 23 could not be transformed: 'return ['
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[23]
✓ Number of created sketches:0
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 11

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 92f3c8fb52b32b79005b4864c31a5c2d8c45f4b1
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_push.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [15]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:0
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 12

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 4c2fc490f280ca3921785d8f58a37274ced35ce6
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/no_command.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_no_command.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/no_command.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [3, 9]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 3
✓
Start transforming line: 9
x Fail! Line: 9 could not be transformed: 'return (command.script_parts'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 1 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[9]
✓ Number of created sketches:1
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 13

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 2af65071d84a7d1d14a4126364d9b4c9b5241f3c
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_branch_exists.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_branch_exists.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_branch_exists.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [9, 10]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 9
x Fail! Line: 9 could not be transformed: 'return ('branch' in command.script'
Start transforming line: 10
x Fail! Line: 10 could not be transformed: 'and "fatal A branch named '" in command.stderr'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[9, 10]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 14

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 183b70c8b8885843efefd2bd4e74dc0a7d42d173
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/shells/test_fish.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/fish.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [10, 11, 12, 13, 14]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 10
x Fail! Line: 10 could not be transformed: 'overridden_aliases = os.environ.get('TF_OVERRIDDEN_ALIASES', '').strip()'
Start transforming line: 11
✓
Start transforming line: 12
x Fail! Line: 12 could not be transformed: 'return [alias.strip() alias in overridden_aliases.split(',')]'
Start transforming line: 13
x Fail! Line: 13 could not be transformed: ''
Start transforming line: 14
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 24 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[10, 12, 13]
✓ Number of created sketches:24
Time needed: 0.08 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 15

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 3a39deb485995e67afb1919972cd1c9aaedf4c32
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_add.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_add.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_add.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [8, 9, 15, 17]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 8
x Fail! Line: 8 could not be transformed: 'return ('did not match any file(s) known to git.' in command.stderr'
Start transforming line: 9
x Fail! Line: 9 could not be transformed: 'and "Did you forget to 'git add'?" in command.stderr)'
Start transforming line: 15
x Fail! Line: 15 could not be transformed: 'r"error pathspec '([^']*)' "'
Start transforming line: 17
x Fail! Line: 17 could not be transformed: ''
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[8, 9, 15, 17]
✓ Number of created sketches:0
Time needed: 0.05 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 16

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: d92765d5df6607cb2f2fb67cee7b63f64ac7aa6b
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/bash.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/shells/test_bash.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/bash.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [9, 10, 10, 11, 12]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 9
✓
Start transforming line: 10
✓
Start transforming line: 11
✓
Start transforming line: 12
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 39 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:39
Time needed: 0.02 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 17

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: f7f0660114a02fe49578ec5684dd02c81042d175
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/bash.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/shells/test_bash.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/shells/bash.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [12, 27, 29, 30, 31]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 12
✓
Start transforming line: 27
x Fail! Line: 27 could not be transformed: '@cache('.bashrc', '.bash_profile')'
Start transforming line: 29
✓
Start transforming line: 30
x Fail! Line: 30 could not be transformed: 'return dict('
Start transforming line: 31
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 31 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[27, 30]
✓ Number of created sketches:31
Time needed: 0.04 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 18

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
x FAIL! Suspicious lines are empty!

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 19

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 959b96cf6ec8cedda05dc58efe0e0f3bd6ed2f4e
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push_force.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_push_force.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_push_force.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [15]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 15
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 3 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:3
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 20

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 0a6a3db65d2fc480c5b2f1135137f34c9f06b742
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/dirty_unzip.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_dirty_unzip.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/dirty_unzip.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [16, 31]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 16
✓
Start transforming line: 31
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 21

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 71dc2666ccf62e653291d9a7a08e2c6c3320425b
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_fix_stash.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_fix_stash.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_fix_stash.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [8, 9]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 8
x Fail! Line: 8 could not be transformed: 'return (command.script.split()[1] == 'stash''
Start transforming line: 9
x Fail! Line: 9 could not be transformed: 'and 'usage' in command.stderr)'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[8, 9]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 22

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: faa7ee603057fa98c25507d30180c055d10d13d4
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/test_types.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [84, 85, 86]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 84
✓
Start transforming line: 85
x Fail! Line: 85 could not be transformed: 'self._cached = [self._cached[0]] + sorted('
Start transforming line: 86
x Fail! Line: 86 could not be transformed: 'commands, key=lambda corrected_command corrected_command.priority)'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 3 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[85, 86]
✓ Number of created sketches:3
Time needed: 0.02 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 23

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 4129ff2717cc6e6fa51d70cc4e6c31d56ef8e2c9
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/utils.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/test_utils.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/utils.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [183]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 183
x Fail! Line: 183 could not be transformed: 'with shelve.open(cache_path) as db'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[183]
✓ Number of created sketches:0
Time needed: 0.0 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 24

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 12394ca8423a438915fed996383b44471fc1139d
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/test_types.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [57, 58, 64, 68, 70]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 57
x Fail! Line: 57 could not be transformed: 'command.script != first.script or \'
Start transforming line: 58
✓
Start transforming line: 64
x Fail! Line: 64 could not be transformed: 'commands = {(command.script, command.side_effect) command'
Start transforming line: 68
x Fail! Line: 68 could not be transformed: 'or command.side_effect != self._cached[0].side_effect}'
Start transforming line: 70
x Fail! Line: 70 could not be transformed: ''
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 12 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[57, 64, 68, 70]
✓ Number of created sketches:12
Time needed: 0.04 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 25

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 42a8b4f639269886e468762e6d100b6f01aad8ab
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/mkdir_p.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_mkdir_p.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/mkdir_p.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [13]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 13
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 3 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:3
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 26

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 7cb0388ed0845545e878b29783bbf8e901a02745
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/vagrant_up.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_vagrant_up.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/vagrant_up.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [10, 13]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 10
x Fail! Line: 10 could not be transformed: 'machine = ""'
Start transforming line: 13
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 4 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[10]
✓ Number of created sketches:4
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 27

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: bc6b107066d3f1e60b4cfcaa8cf6399e98cf1b1c
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/open.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_open.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/open.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [26]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 26
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:0
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 28

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 88831c424f569e6a55fc98883d3eeecc7d425b18
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/fix_file.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_fix_file.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/fix_file.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [23, 26, 28, 64, 66]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 23
x Fail! Line: 23 could not be transformed: ''^{file} \(line {line}\)','
Start transforming line: 26
x Fail! Line: 26 could not be transformed: '   '
Start transforming line: 28
x Fail! Line: 28 could not be transformed: '   '
Start transforming line: 64
x Fail! Line: 64 could not be transformed: 'editor_call = '{} {} +{}'.format(os.environ['EDITOR'],'
Start transforming line: 66
x Fail! Line: 66 could not be transformed: ' m.group('line'))'
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 0 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[23, 26, 28, 64, 66]
✓ Number of created sketches:0
Time needed: 0.03 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 29

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 4a2f869c6d6ef03d8d7ada1121cc6631d7ef979e
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/test_types.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/types.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [26, 27, 28]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 26
x Fail! Line: 26 could not be transformed: '"""Returns new settings with new values from `kwargs`."""'
Start transforming line: 27
✓
Start transforming line: 28
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 11 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[26]
✓ Number of created sketches:11
Time needed: 0.03 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 30

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: de513cacb150049e3f95434f8d6d30b7ed1e0ea7
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/fix_file.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_fix_file.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/fix_file.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [53]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 53
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 6 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:6
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 31

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: 66e2ec7e3f0d3f848c01d87bb3503b0ff90fc78a
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_diff_staged.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_git_diff_staged.py
TEST EXECUTION:
x FAIL! Test fails or has errors.
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/git_diff_staged.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [13]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 13
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 3 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:3
Time needed: 0.01 minutes.

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+

                                          BUG: 32

+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+*+
GIT RESET TO: cb33c912e5f2f4c2da6b70d708ff0437bfcd3b94
bug_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/ls_lah.py
bug_test_file = /Users/daniel/Bachelorarbeit/Projects/Thefuck/tests/rules/test_ls_lah.py
TEST EXECUTION:
✓ SUCCESSFUL! TEST PASSES
✓ PySketchFix is started...
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

	                              PySketchFix®: Sketches are created.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
PySketchFix will run with:
bug file: /Users/daniel/Bachelorarbeit/Projects/Thefuck/thefuck/rules/ls_lah.py
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The suspicious lines which are taken are: [2]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start transforming line: 2
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Start creating 5 sketches.
✓
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x Lines where no transformation is found:[]
✓ Number of created sketches:5
Time needed: 0.01 minutes.
#######################################################################################################
Results with bug numbers:
Project:Thefuck
Failing:[1, 2, 5, 6, 7, 9, 10, 11, 12, 13, 15, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31]
Passing:[3, 4, 14, 16, 17, 32]
Empty Suspicious Lines:[18]
No bug information is found:[8]
